packages/runtime/src/runtimes/react/controls/list.tsx

```diff
@@ -11,17 +12,26 @@ type ListControlValueProps<T extends ListControlDefinition> = {
   definition: T
   data: ListControlData<T> | undefined
   children(value: ListControlValue<T>): JSX.Element
+  control: ListControl
 }
 
 export function ListControlValue<T extends ListControlDefinition>({
   definition,
   data,
   children,
+  control,
 }: ListControlValueProps<T>): JSX.Element {
+  // Create new controls every time the user adds or removes an item.
+  const itemsControl = useMemo(() => control?.setItemsControl(data), [data?.length, control])
```

Comment:
How come we're doing this here instead of handling it in the control? It seems very weird to let this logic "spill out", it breaks encapsulation

Reply:
Don't you find it weird you're calling `setItemsControl` in `useMemo`, mutating state on a "getter"?

Reply:
Yeah, I also didn't like it. We could discuss this further.

Reply:
So right now we're creating the control from an action. Same as with the builder, we need to have a `subscribeToValue` function available to the `createPropController`. 

The problem is, currently, we don't have any way to subscribe to a value from inside an action. I don't think there's any official way to do `store.subscribe` from an action.

One way to solve this is to have a custom thunk that also passes the `store.subscribe` as the third argument for the action, similar to this [package](https://github.com/AugurProject/redux-thunk-subscribe/blob/master/src/index.js).

But I'm not 100% sure that would be the best approach.

Reply:
Decided to use useEffect for now. Added comments there as to how we can improve it.